templates/editentity/arp.php

```diff
@@ -31,14 +33,37 @@ function attributeNameValuePairId($arpAttrName, $value)
                 $arpAttributeUsed = $arp !== null && array_key_exists($attribute['name'], $arp);
                 $arpSpecifyValues = (isset($attribute['specify_values']) && $attribute['specify_values']);
                 $arpAttrName = htmlentities($attribute['name'], ENT_QUOTES, "UTF-8");
+                $selectedSource = $arpAttributeHelper->getSelectedSource($arp[$attribute['name']]);
+                if (isset($attribute['source']) && !empty($attribute['source'])) {
+                    $arpAttrSource = htmlentities($attribute['source'], ENT_QUOTES, "UTF-8");
+                }
                 ?>
                 <tr class="attribute_select_row">
                     <td>
                         <label title="<?php echo $arpAttrName ?>"><?php echo htmlentities($label, ENT_QUOTES, "UTF-8"); ?></label>
                     </td>
+                    <td>
+                        <select name="arp_attribute_source[<?php echo $arpAttrName ?>]">
+                            <?php
+                            foreach ($arpAttributeSources as $arpAttrSource): ?>
+                            <option
+                                value="<?php echo htmlentities($arpAttrSource, ENT_QUOTES, "UTF-8"); ?>"
+                                <?php if ($selectedSource == $arpAttrSource): ?>
+                                selected
+                                <?php endif; ?>
+                            >
+
+                                <?php echo $arpAttrSource; ?>
+                            </option>
+                            <?php endforeach; ?>
+                        </select>
+                    </td>
                     <td data-specify-values="<?php echo $arpSpecifyValues ? 'true' : 'false'; ?>">
                         <?php if ($arpAttributeUsed): ?>
-                            <?php foreach ($arp[$attribute['name']] as $value): ?>
+                            <?php
+                            foreach ($arp[$attribute['name']] as $value):
+                                $value = $arpAttributeHelper->getAttributeFilterValue($value);
```

Comment:
I understand you didn't introduce this problem, but this looks like XSS. Also, it's better to escape the value right where you print it. So, always do something like 'print escape($x)', never 'print $x'.